src/story.ts

```diff
@@ -0,0 +1,29 @@
+export type StoryId = string;
+export type StoryKind = string;
+export type StoryName = string;
+
+export type StoryReturnType<T> = T extends (storyFn: infer StoryFn, context: any) => any
+  ? StoryFn extends (...args: any) => any
```

Comment:
@wKich what is the benefit of this nested conditional? I'm asking because I'm getting a strange error when I try to use these types:

```ts
import { KindMeta, StoryMeta } from './csfTypes';

export const basicBtn: StoryMeta = () => <Button>hello</Button>;
```
`Type 'Element' is not assignable to type 'never'.ts(2322)`
`csfTypes.d.ts(38, 18): The expected type comes from the return type of this signature.`

Reply:
Thanks for review. But according by my first comment for PR, you don't use this typings directly. Only from `@storybook/<framework>` module. So after success review I'll add typings for all support frameworks in Storybook main repo.

I think, I should remove `Component` generic fallback to `unknown`, to be meta types more strict.